src/Functions/tokens.cpp

```diff
@@ -112,10 +112,26 @@ class ExecutableFunctionTokens : public IExecutableFunction
         if (input_rows_count == 0)
             return ColumnArray::create(std::move(col_result), std::move(col_offsets));
 
-        if (const auto * column_string = checkAndGetColumn<ColumnString>(col_input.get()))
-            executeImpl(*token_extractor, *column_string, *col_offsets, input_rows_count, *col_result);
-        else if (const auto * column_fixed_string = checkAndGetColumn<ColumnFixedString>(col_input.get()))
-            executeImpl(*token_extractor, *column_fixed_string, *col_offsets, input_rows_count, *col_result);
+        const auto execute_with_tokenizer = [&](const ITokenExtractor & extractor)
```

Comment:
We will expand the logics of this `execute_with_tokenizer` very soon, so please make this a real function to not obscure the code of `executeImpl`. (This will also avoid the lambda ;p)